docs/conf.py

```diff
@@ -0,0 +1,336 @@
+    "colon_fence",
+    "dollarmath",
+    "substitution",
+    "tasklist",
+]
+
+myst_heading_anchors = 4
+suppress_warnings = ["etoc.toctree"]
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ["_templates"]
+
+# List of patterns, relative to source directory, that match files and
+# directories to ignore when looking for source files.
+# This pattern also affects html_static_path and html_extra_path.
+exclude_patterns = [
+    "_build",
+    "Thumbs.db",
+    ".DS_Store",
+    ".jupyter_cache",
+    "jupyter_execute",
+]
+
+# -- Examples gallery scrapers -------------------------------------------------
+
+
+class VideoScraper:
+    """Video file scraper class.
+    Scrapes video files that were saved to disk by the example scripts.
+    Based on the sphinx example scraper "Example 2: detecting image files on disk"
+    https://sphinx-gallery.github.io/stable/advanced.html#example-2-detecting-image-files-on-disk
+    """
+
+    def __init__(self):
+        self.seen = set()
+
+    def __repr__(self):
+        return "VideoScraper"
+
+    def __call__(self, block, block_vars, gallery_conf):
+        """Video file scraper.
+        Scrapes video files that were saved to disk by the example scripts.
+        Based on the sphinx example scraper "Example 2: detecting image files on disk"
+        https://sphinx-gallery.github.io/stable/advanced.html#example-2-detecting-image-files-on-disk
+        """
+        # Find all video files in the directory of this example.
+        video_file_extensions = [".mp4", ".mov"]
+        path_current_example = os.path.dirname(block_vars["src_file"])
+        video_paths = sorted(
+            str(p.resolve())
+            for p in Path(path_current_example).glob("**/*")
+            if p.suffix in video_file_extensions
+        )
+
+        # Iterate through the videos, copy them to the sphinx-gallery output directory
+        video_names = []
+        image_path_iterator = block_vars["image_path_iterator"]
+        rst = ""
+        for video in video_paths:
+            if video not in self.seen:
+                self.seen |= set(video)
+                this_video_path = image_path_iterator.next()
+                this_video_path = (
+                    os.path.splitext(this_video_path)[0]
+                    + os.path.splitext(video)[1]
+                )
+                video_names.append(this_video_path)
+                shutil.move(video, this_video_path)
+                self.video_thumbnail(this_video_path)
+                rst += self.rst_video_template(this_video_path)
+
+        # Use the `figure_rst` helper function to generate reST for image files
+        # result = figure_rst(video_names, gallery_conf['src_dir'])
+        return rst
+
+    def video_thumbnail(self, video_filename):
+        """Save PNG screenshot of the first video frame."""
+        reader = iio.get_reader(video_filename)
+        first_frame = reader.get_data(0)
```

Comment:
Note that the image is from the first frame of the video.

That means in some cases it isn't a very interesting picture. Like the layer planes example, which fades in from black, so you don't see anything very exciting in the thumbnail image.

I tried to make this function choose a frame from the middle of the videos, but ffmpeg is returning "Inf" lengths for these movie files, so I have no good way to know where the halfway point is.